### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

File: mc-conflicts-ui/src/conflicts/models/conflictStore.ts [hunk 1]
```diff
@@ -89,15 +77,19 @@ export const ConflictStore = types
...
         const conflicts = result.data.data;
         resetSelectedConflict();
         self.conflicts.replace(conflicts.map(conflictFormatter));
-        self.state = 'done';
+        self.pagination.setTotalItems(result.data.total)
+        self.state = ResponseState.done;
       } catch (error) {
-        self.state = 'error';
+        console.error(error);
+        self.state = ResponseState.error;
```
Comment (syncush) [inline mc-conflicts-ui/src/conflicts/models/conflictStore.ts]: `ResponseState.ERROR`

File: mc-conflicts-ui/src/conflicts/models/rootStore.ts [hunk 2]
```diff
@@ -1,10 +1,11 @@
-import { types, Instance, getEnv } from "mobx-state-tree";
+import { types, Instance, getEnv, onAction } from "mobx-state-tree";
 import { useContext, createContext } from 'react';
 import { ConflictStore } from './conflictStore';
+import { ResponseState } from "../../common/models/responseState";
```
Comment (syncush) [inline mc-conflicts-ui/src/conflicts/models/rootStore.ts]: `import { ResponseState } from "../../common/models/ResponseState";`

File: mc-conflicts-ui/src/conflicts/models/rootStore.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 
 export const baseRootStore = types
   .model({
-    conflictsStore: types.optional(ConflictStore, { state: 'pending', searchParams: {} }),
+    conflictsStore: types.optional(ConflictStore, { state: ResponseState.pending, searchParams: {} }),
```
Comment (syncush) [inline mc-conflicts-ui/src/conflicts/models/rootStore.ts]: `ResponseState.PENDING`

**Review by syncush — CHANGES_REQUESTED** [review]

File: mc-conflicts-ui/src/common/models/responseState.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export enum ResponseState {
+  "pending" = "pending",
+  "done" = "done",
+  "error" = "error",
+}
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/common/models/responseState.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx [hunk 5]
```diff
@@ -46,13 +48,13 @@ export const ConflictsTable: React.FC = observer(() => {
   const classes = useStyle();
   const { conflictsStore } = useStore();
 
-  if (conflictsStore.state === "pending") {
+  if (conflictsStore.state === ResponseState.pending) {
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx [hunk 6]
```diff
@@ -46,13 +48,13 @@ export const ConflictsTable: React.FC = observer(() => {
...
     return (
       <div className={classes.infoContainer}>
         <CircularProgress className={classes.infoContent} />
       </div>
     );
-  } else if (conflictsStore.state === "error") {
+  } else if (conflictsStore.state === ResponseState.error) {
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/models/conflictStore.ts [hunk 7]
```diff
@@ -78,7 +66,7 @@ export const ConflictStore = types
       conflictResponse
     > {
       self.conflicts = cast([]);
-      self.state = 'pending';
+      self.state = ResponseState.pending;
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/conflictStore.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/models/conflictStore.ts [hunk 8]
```diff
@@ -89,15 +77,19 @@ export const ConflictStore = types
...
+      params.limit = self.pagination.itemsPerPage;
 
       try {
         const result = yield self.root.fetch('/conflicts', params);
         const conflicts = result.data.data;
         resetSelectedConflict();
         self.conflicts.replace(conflicts.map(conflictFormatter));
-        self.state = 'done';
+        self.pagination.setTotalItems(result.data.total)
+        self.state = ResponseState.done;
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/conflictStore.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/models/conflictStore.ts (hunk 1 above)
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/conflictStore.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/models/rootStore.ts (hunk 2 above)
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/rootStore.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/models/rootStore.ts (hunk 3 above)
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/rootStore.ts]: fixed in 51f8b14

File: mc-conflicts-ui/src/conflicts/components/conflict-item.tsx [hunk 9]
```diff
@@ -1,14 +1,12 @@
 import React from 'react';
-import { IConflict } from '../models/conflictStore';
 import Typography from '@material-ui/core/Typography';
+import { IConflict } from '../models/conflict';
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/components/conflict-item.tsx]: fixed in d35869a

File: mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx [hunk 10]
```diff
@@ -4,9 +4,11 @@ import { observer } from "mobx-react-lite";
 import CircularProgress from "@material-ui/core/CircularProgress";
 import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
 import ConflictItem from "./conflict-item";
-import { IConflict } from "../models/conflictStore";
 import { Typography } from "@material-ui/core";
 import {CellMetadata, SmartTable} from 'mc-react-components'
+import { IConflict } from "../models/conflict";
+import { ResponseState } from "../../common/models/responseState";
+
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/components/conflicts-table.tsx]: fixed in d35869a

File: mc-conflicts-ui/src/conflicts/models/conflictStore.ts [hunk 11]
```diff
@@ -13,6 +13,9 @@ import { feature } from '@turf/helpers';
 import { Feature, Geometry } from 'geojson';
 import { ConflictSearchParams } from './conflict-search-params';
 import { IRootStore } from './rootStore';
+import { pagination } from './pagination';
+import { ResponseState } from '../../common/models/responseState';
+import { Conflict, IConflict } from './conflict';
```
[the remark this answers is not in the document]
Comment (CptSchnitz) [reply mc-conflicts-ui/src/conflicts/models/conflictStore.ts]: fixed in d35869a

**Review by syncush — APPROVED** [review]

**Pull request merged**